File: leetcode/229-majority-element-ii/majority_element.py
# LC 229: Majority Element II

class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        nums.sort()
        res, n = [], len(nums)

        i = 0
        while i < n:
            j = i + 1
            while j < n and nums[i] == nums[j]:
                j += 1

            if (j - 1) > n // 3:
                res.append(nums[i])
            i = j

        return res

File: leetcode/229-majority-element-ii/MajorityElement.java
// LC 229: Majority Element II

class MajorityElement {
	public List<Integer> majorityElement(int[] nums) {
        Arrays.sort(nums);
        List<Integer> res = new ArrayList<>();
        int n = nums.length;

        int i = 0;
        while (i < n) {
            int j = i + 1;
            while (j < n && nums[i] == nums[j])
                j++;
            
            if (j - i > n /3)
                res.add(nums[i]);
            
            i = j;
        }

        return res;

	}
}
